src/main/java/io/github/mountainrange/mule/managers/KeyBindManager.java

```diff
@@ -21,45 +24,173 @@ public void add(GameState state, KeyCode toAdd, KeyFunction lambda) {
-		}
+		// Buy Land for Player 1
+		toBind.add(new GameState(GameType.SIMULTANEOUS, MULE.PLAY_SCENE, 0), KeyCode.SPACE,
+				   (a) -> {
+					   if (Config.getInstance().numOfPlayers > 0) {
+						   a.manager.buyTile(a.manager.getPlayerList().get(0));
+					   }
+					   return "Bought land for Player 1"; });
 
-		@Override
-		public boolean equals(Object other) {
-			if (!(other instanceof GameState)) {
-				return false;
-			} else if (other == this) {
-				return true;
-			} else {
-				GameState toCompare = (GameState) other;
-				return toCompare.gameType.equals(this.gameType) && toCompare.sceneName.equals(this.sceneName);
-			}
-		}
-	}
+		// Buy Land for Player 2
+		toBind.add(new GameState(GameType.SIMULTANEOUS, MULE.PLAY_SCENE, 0), KeyCode.P,
+				   (a) -> {
+					   if (Config.getInstance().numOfPlayers > 1) {
+						   a.manager.buyTile(a.manager.getPlayerList().get(1));
+					   }
+					   return "Bought land for Player 2"; });
 
-	public class KeyBindPackage {
-		public GameManager manager;
+		// Buy Land for Player 3
+		toBind.add(new GameState(GameType.SIMULTANEOUS, MULE.PLAY_SCENE, 0), KeyCode.Q,
+				   (a) -> {
+					   if (Config.getInstance().numOfPlayers > 2) {
+						   a.manager.buyTile(a.manager.getPlayerList().get(2));
+					   }
+					   return "Bought land for Player 3"; });
+
+		// Buy Land for Player 3
+		toBind.add(new GameState(GameType.SIMULTANEOUS, MULE.PLAY_SCENE, 0), KeyCode.PERIOD,
+				   (a) -> {
+					   if (Config.getInstance().numOfPlayers > 3) {
+						   a.manager.buyTile(a.manager.getPlayerList().get(3));
+					   }
+					   return "Bought land for Player 4"; });
 
-		public KeyBindPackage(GameManager m) {
-			this.manager = m;
-		}
-	}
 
-	public interface KeyFunction {
-		public String act(KeyBindPackage m);
+		// ----------------------------Hotseat Hacks-------------------------------
+
+		toBind.add(new GameState(GameType.HOTSEAT, MULE.PLAY_SCENE, 0), KeyCode.X,
+				   (a) -> {
+					   a.manager.commentYourCodeGuys();
+					   return "Bought land for Player 4"; });
+
+
+
+		// ----------------------------Buying Tiles-------------------------------
+
+		toBind.add(new GameState(GameType.SIMULTANEOUS, MULE.PLAY_SCENE, 0), KeyCode.SPACE,
+				   (a) -> {
+					   a.manager.buyTile();
+					   return "Bought Tile"; });
+		toBind.add(new GameState(GameType.HOTSEAT, MULE.PLAY_SCENE, 0), KeyCode.SPACE,
+				   (a) -> {
+					   a.manager.buyTile();
+					   return "Bought Tile"; });
+		toBind.add(new GameState(GameType.SIMULTANEOUS, MULE.PLAY_SCENE, 1), KeyCode.SPACE,
+				   (a) -> {
+					   a.manager.buyTile();
+					   return "Bought Tile"; });
+		toBind.add(new GameState(GameType.HOTSEAT, MULE.PLAY_SCENE, 1), KeyCode.SPACE,
+				   (a) -> {
+					   a.manager.buyTile();
+					   return "Bought Tile"; });
+
+
+		// ----------------------------Movement Keys-------------------------------
+		// Moving Up
+		toBind.add(new GameState(GameType.HOTSEAT, MULE.PLAY_SCENE, 0), KeyCode.UP,
```

Comment:
Actually, we could just pass in a list of values, and the add function can iterate over the list adding things. That would keep all these entries, but make input way easier (and less error prone).

I'm going to do that if no one objects...